tests/test_cli.py

```diff
@@ -39,3 +43,110 @@ def test_show_external_dependencies_not_exists(monkeypatch):
     sh    ✓
 """
     )
+
+
+@pytest.mark.parametrize(
+    "params",
+    (
+        pytest.param(["--help"], id="alone"),
+        pytest.param(
+            ["--verbose", "--extract-dir", "unblob", "--depth", "2", "--help", "tests"],
+            id="eager_1",
+        ),
+        pytest.param(
+            ["--verbose", "--extract-dir", "unblob", "--depth", "2", "tests", "--help"],
+            id="eager_2",
+        ),
+    ),
+)
+def test_help(params):
+    runner = CliRunner()
+    result = runner.invoke(unblob.cli.cli, params)
+    assert result.exit_code == 0
+    # NOTE: In practive, it writes "Usage: unblob ...", this is done in the `cli.main` with `click.make_context`
+    assert result.output.startswith("Usage: cli [OPTIONS] [FILES]...")
+
+
+@pytest.mark.parametrize(
+    "params",
+    (
+        pytest.param(["-v"], id="v"),
+        pytest.param(["--verbose"], id="verbose"),
+        pytest.param(["-e", "unblob"], id="e"),
+        pytest.param(["--extract-dir", "unblob"], id="extract-dir"),
+        pytest.param(["-d", "2"], id="d"),
+        pytest.param(["--depth", "2"], id="depth"),
+        pytest.param(
+            ["--verbose", "--extract-dir", "unblob", "--depth", "2"],
+            id="verbose+extract-dir+depth",
+        ),
+    ),
+)
+def test_without_file(params: List[str]):
+    runner = CliRunner()
+    result = runner.invoke(unblob.cli.cli, params)
+    assert result.exit_code == 0
+    assert "No files given, doing nothing!" in result.output
+
+
+def test_empty_dir_as_file(tmp_path: Path):
+    runner = CliRunner()
+    out_path = tmp_path.joinpath("out")
+    out_path.mkdir()
+    in_path = tmp_path.joinpath("in")
+    in_path.mkdir()
+    result = runner.invoke(
+        unblob.cli.cli, ["--extract-dir", str(out_path), str(in_path)]
+    )
+    assert result.exit_code == 0
+
+
+@pytest.mark.parametrize(
+    "params",
+    (
+        pytest.param([], id="empty"),
+        pytest.param(["--verbose"], id="verbose"),
+        pytest.param(["--depth", "2"], id="depth"),
+        pytest.param(["--verbose", "--depth", "2"], id="verbose+depth"),
+    ),
+)
+def test_archive_success(params, tmp_path: Path):
+    runner = CliRunner()
+    in_path = (
+        Path(__file__).parent
+        / "integration"
+        / "archive"
+        / "zip"
+        / "regular"
+        / "__input__/"
+    )
+    new_params = params + ["--extract-dir", str(tmp_path), str(in_path)]
+    result = runner.invoke(unblob.cli.cli, new_params)
```

Comment:
I'd mock the `process_file` call and assert on its arguments. Wouldn't run this with any of the real integration test assets.